arch/arm/boot/dts/overlays/revpi-connect4-overlay.dts

```diff
@@ -0,0 +1,262 @@
+
+	fragment@0 {
+		target-path = "/";
+		__overlay__ {
+			compatible = "kunbus,revpi-connect4", "brcm,bcm2711";
+
+			pibridge {
+				pinctrl-names = "default";
+				compatible = "kunbus,pibridge";
+				pinctrl-0 = <&digital_io_pins>;
+
+				/* digital IN, digital OUT */
+				connect-gpios = <&gpio 6 GPIO_ACTIVE_HIGH>,
+						<&gpio 16 GPIO_ACTIVE_HIGH>;
+
+				left-sniff-gpios = <&expander_core 8 GPIO_ACTIVE_HIGH>,
+						   <&expander_core 9 GPIO_ACTIVE_HIGH>;
+				right-sniff-gpios = <&expander_core 11 GPIO_ACTIVE_HIGH>,
+						   <&expander_core 12 GPIO_ACTIVE_HIGH>;
+			};
+
+			leds {
+				compatible = "gpio-leds";
+				power_red {
+					label = "power:1:fault";
+					gpios = <&expander_core 0 GPIO_ACTIVE_HIGH>;
+					linux,default-trigger = "power_red";
+				};
+				a1_red {
+					label = "a1:red:status";
+					gpios = <&expander_core 2 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_red";
+				};
+				a1_green {
+					label = "a1:green:status";
+					gpios = <&expander_core 1 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_green";
+				};
+				a1_blue {
+					label = "a1:blue:status";
+					gpios = <&expander_core 3 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_blue";
+				};
+				a2_red {
+					label = "a2:red:status";
+					gpios = <&expander_core 5 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_red";
+				};
+				a2_green {
+					label = "a2:green:status";
+					gpios = <&expander_core 4 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_green";
+				};
+				a2_blue {
+					label = "a2:blue:status";
+					gpios = <&expander_core 6 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_blue";
+				};
+				a3_red {
+					label = "a3:red:status";
+					gpios = <&expander_power 1 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_red";
+				};
+				a3_green {
+					label = "a3:green:status";
+					gpios = <&expander_power 0 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_green";
+				};
+				a3_blue {
+					label = "a3:blue:status";
+					gpios = <&expander_power 2 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_blue";
+				};
+			};
+		};
+	};
+
+	/* PINMUX */
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			pb_uart_pins: pb_uart_pins {
+				/* TX RX TX_EN */
+				brcm,pins     = <4 5 7>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			digital_io_pins: digital_io_pins {
+				/* digital IN, digital OUT */
+				brcm,pins     = <6 16>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN
+				                 BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF
+				                 BCM2835_PUD_OFF>;
+			};
+			exp_core_pins: exp_core_pins {
+				brcm,pins     = <8>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			exp_power_pins: exp_power_pins {
+				brcm,pins     = <9>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			i2c1_pins: i2c1_pins {
```

Comment:
As the i2c1_pins are already defined from the bcm2711-rpi-cm4.dtsi this name will conflict with that definition. You could modify the original definition like in the core overlay:
https://github.com/RevolutionPi/linux/blob/96fdda2f0d71e10f7e5c67aef155cf73a07d0e50/arch/arm/boot/dts/overlays/revpi-core-overlay.dts#L123
or you could think of a new name for the pins.

Reply:
or it should be removed, as the definition of i2c1_pins in bcm2711-rpi-cm4.dts is what we want.

Reply:
No, we decided to do our pinmuxing independent of the dtsi.

Reply:
then maybe it's more proper to use a new name for it. Suggestion: "revpi-i2c1-pins", "connect-i2c1-pins", "conn-i2c1-pins". 

Btw, for the consistency, would it be needed to apply the same rule for pinmuxing of spi1 and rs485, e.g., "conn-spi1-pins", "conn-rs485-pins"

Reply:
Or you can modify the original pinmuxes like the in the core overlay

Reply:
are you mean to delete the original node with "/delete-node/"?  what if the original i2c1_pins do not exist in the future.

Reply:
https://github.com/RevolutionPi/linux/blob/96fdda2f0d71e10f7e5c67aef155cf73a07d0e50/arch/arm/boot/dts/overlays/revpi-core-overlay.dts#L123
```
	fragment@2 {
		target = <&i2c1_pins>;
		__overlay__ {
			/* sda scl */
			brcm,pins     = <44 45>;
			brcm,function = <BCM2835_FSEL_ALT2>;
			brcm,pull     = <BCM2835_PUD_OFF>;
		};
	};
```